task4.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <fstream>
+using namespace std;
+const int MAX_SIZE = 25;
+const int BUFFER_SIZE = 1024;
+struct Pair {
+	int x1, x2;
+};
+struct Relation {
+	Pair relationSet[MAX_SIZE];
+};
+void createPair(Pair* pair,const int& x1,const int& x2) {
+	pair->x1 = x1;
+	pair->x2 = x2;
+}
+void addPairToRelation(Pair& pair, Relation (&set)[MAX_SIZE],int& i) {
+	set->relationSet[i] = pair;
+}
+Pair* readPairFromFile(ifstream& in) {
+		char buffer[BUFFER_SIZE];
+		in.getline(buffer, BUFFER_SIZE);
+		Pair* pair = new Pair;
+		createPair(pair, (int)buffer[0]-48, (int)buffer[2]-48);
+		return pair;
+}
+void  readRelationFromFile(ifstream& in,Relation (&set)[MAX_SIZE],int& i) {
```

Comment:
why & on set? Also, why you pass ifstream as & when you will open it inside the func?